src/adapters/surrealDB/pipeline/postprocess/query/cleanQueryRes.ts

```diff
@@ -0,0 +1,95 @@
+import type { PipelineOperation, BQLResponseSingle, BormConfig } from '../../../../../types';
+import { produce } from 'immer';
+import type { SurrealDbResponse, EnrichedBqlQuery } from '../../../types/base';
+
+const processNull = (config: BormConfig, obj: Record<string, unknown>) => {
+	for (const key in obj) {
+		const value = obj[key];
+
+		if (config.query?.returnNulls) {
+			if (Array.isArray(value) && value.length === 0) {
+				obj[key] = null;
+				continue;
+			}
+		} else {
+			if (value === null) {
+				delete obj[key];
+				continue;
+			}
+
+			if (Array.isArray(value) && value.length === 0) {
+				delete obj[key];
+				continue;
+			}
+		}
+	}
+};
+
+const cleanUpObj = ({
+	config,
+	item,
+	query,
+}: {
+	query: EnrichedBqlQuery;
+	config: BormConfig;
+	item: BQLResponseSingle;
+}) => {
+	processNull(config, item);
+
+	// INTERNAL SYMBOLS
+	Object.getOwnPropertySymbols(item).forEach((symbol) => {
+		delete item[symbol];
+	});
+
+	/// USER FACING METADATA
+	if (config.query?.noMetadata === true) {
+		// eslint-disable-next-line no-param-reassign
+		Object.keys(item).forEach((k: string) => {
+			if (k.startsWith('$')) {
+				delete item[k];
+			}
+		});
+	}
+
+	// filter out id
+	if (query.$idNotIncluded) {
+		delete item['id'];
```

Comment:
Btw there is an issue in the typeDB queries is that the idField might not be "id" 

The idField is specified in the borm schema and could be any field. Obviously most of the time is "id" but it could be named differently so we need to get schema.idFields[0] instead (in the future this will be updated so it might work with compound ids)

Reply:
No problem, handled idFields now